export-distro fails to be killed with ctrl-c after make run

To reproduce on current master build the binaries with `make build`, DON'T run mongodb (i.e. turn it off or disable it), and do:

```bash
make run
# wait a few seconds for things to try to connect and go into a connection loop
# interrupt with ctrl-c
```

You will continue to see export-distro show errors:

```
^Clevel=WARN ts=2019-05-03T02:05:40.613376466Z app=edgex-sys-mgmt-agent source=main.go:71 msg="terminating: interrupt"
level=WARN ts=2019-05-03T02:05:40.613389706Z app=edgex-support-notifications source=main.go:75 msg="terminating: interrupt"
level=WARN ts=2019-05-03T02:05:40.613503134Z app=edgex-core-command source=main.go:71 msg="terminating: interrupt"
Makefile:70: recipe for target 'run' failed
make: *** [run] Interrupt

me@localhost:~/go/src/github.com/edgexfoundry/edgex-go$ ^C
me@localhost:~/go/src/github.com/edgexfoundry/edgex-go$ ^C
me@localhost:~/go/src/github.com/edgexfoundry/edgex-go$ level=ERROR ts=2019-05-03T02:05:41.527711086Z app=edgex-export-distro source=client.go:28 msg="Error getting all registrations: http://localhost:48071/api/v1/registration. Error: Get http://localhost:48071/api/v1/registration: dial tcp 127.0.0.1:48071: connect: connection refused"
level=INFO ts=2019-05-03T02:05:41.535249365Z app=edgex-export-distro source=registrations.go:287 msg="Waiting for client microservice"
^C
me@localhost:~/go/src/github.com/edgexfoundry/edgex-go$ level=ERROR ts=2019-05-03T02:05:42.539137539Z app=edgex-export-distro source=client.go:28 msg="Error getting all registrations: http://localhost:48071/api/v1/registration. Error: Get http://localhost:48071/api/v1/registration: dial tcp 127.0.0.1:48071: connect: connection refused"
level=INFO ts=2019-05-03T02:05:42.544166385Z app=edgex-export-distro source=registrations.go:287 msg="Waiting for client microservice"
```

--- a/cmd/export-distro/main.go
+++ b/cmd/export-distro/main.go
@@ -59,7 +59,7 @@ func main() {
 
 	listenForInterrupt(errs)
 
-	distro.Loop()
+	go distro.Loop()
 
 	// Time it took to start service
 	distro.LoggingClient.Info("Service started in: " + time.Since(start).String())

--- a/internal/export/distro/init.go
+++ b/internal/export/distro/init.go
@@ -40,6 +40,7 @@ var registryUpdates chan interface{} //A channel for "config updates" sourced fr
 var messageClient messaging.MessageClient
 var messageErrors chan error
 var messageEnvelopes chan *msgTypes.MessageEnvelope
+var processStop chan bool
 
 func Retry(useRegistry bool, useProfile string, timeout int, wait *sync.WaitGroup, ch chan error) {
 	until := time.Now().Add(time.Millisecond * time.Duration(timeout))
@@ -89,6 +90,8 @@ func Init(useRegistry bool) bool {
 
 	var err error
 	messageErrors, messageEnvelopes, err = initMessaging(messageClient)
+	processStop = make(chan bool)
+
 	if err != nil {
 		LoggingClient.Error(err.Error())
 		return false
@@ -106,6 +109,11 @@ func Destruct() {
 
 	if registryUpdates != nil {
 		close(registryUpdates)
+	}
+
+	if processStop != nil {
+		processStop <- true
+		close(processStop)
 	}
 
 	if messageErrors != nil {

--- a/internal/export/distro/registrations.go
+++ b/internal/export/distro/registrations.go
@@ -292,6 +292,9 @@ func Loop() {
 				LoggingClient.Error(fmt.Sprintf("with error: %s", err.Error()))
 			}
 			return
+		case <-processStop:
+			LoggingClient.Info(fmt.Sprintf("received term signal"))
+			return
 		case <-time.After(time.Second):
 		}
 		allRegs, err = getRegistrations()
@@ -311,14 +314,14 @@ func Loop() {
 		select {
 		case e := <-messageErrors:
 			// kill all registration goroutines
-			for k, reg := range registrations {
-				if !reg.deleteFlag {
-					// Do not write in channel that will not be read
-					reg.chRegistration <- nil
-				}
-				delete(registrations, k)
-			}
+			stop(registrations)
 			LoggingClient.Error(fmt.Sprintf("exit msg: %s", e.Error()))
+			return
+
+		case <-processStop:
+			// kill all registration goroutines
+			stop(registrations)
+			LoggingClient.Info(fmt.Sprintf("received term signal"))
 			return
 
 		case update := <-registrationChanges:
@@ -350,5 +353,15 @@ func Loop() {
 				}
 			}
 		}
+	}
+}
+
+func stop(registrations map[string]*registrationInfo) {
+	for k, reg := range registrations {
+		if !reg.deleteFlag {
+			// Do not write in channel that will not be read
+			reg.chRegistration <- nil
+		}
+		delete(registrations, k)
 	}
 }

